app/src/lib/stores/app-store.ts

```diff
@@ -2384,16 +2385,28 @@ export class AppStore extends TypedBaseStore<IAppState> {
           // look up if the tracked branch matches a protected remote branch
           const { upstreamWithoutRemote } = gitStore.tip.branch
 
-          if (upstreamWithoutRemote !== null) {
-            const isProtected = await this.repositoriesStore.isBranchProtected(
+          // ensure the remote associated with this branch is the one we
+          // are checking the branch protections against
+          const remoteURL = gitStore.currentRemote.url
+
+          if (
+            upstreamWithoutRemote !== null &&
+            urlMatchesCloneURL(remoteURL, repository.gitHubRepository)
+          ) {
+            const {
+              branchProtectionsFound,
+              isRemoteBranchProtected,
+            } = await this.repositoriesStore.getBranchProtectionContext(
               repository.gitHubRepository,
               upstreamWithoutRemote
             )
 
-            if (isProtected) {
-              log.debug(
-                `[_commitIncludedChanges] the upstream ref '${upstreamWithoutRemote}' is protected by the GitHub API`
-              )
+            if (branchProtectionsFound) {
+              this.statsStore.recordCommitWithBranchProtectionsEnabled()
```

Comment:
I believe that for this metric for be able to fulfil the objectives outlined in the issue, in particular the one quoted below, we're going to have to track this regardless of whether the current branch is published or not. The condition should simply be whether the repository has *any* protected branches or not.

> On repos with protected branches, % of commits to those branches

Reply:
@niik I'll have a go at rejigging this now that @nerdneha also followed up. I encountered some scenarios where working in forks meant it felt more reliable to look at the tracking information, but I'll see if I can make this simpler to just rely on `repository.githubRepository`

Reply:
this is addressed in 006d484 correct?

Reply:
Actually, I _think_ this is covered by 6520ce892c86cd65434341e5c5bb056284b219be

Reply:
ah yes, that's right! thank you!